### PR TITLE
Add more docs for piping sources

### PR DESCRIPTION
**Pull request by Gedochao**

### CODE REVIEW

**Review by lwronski — APPROVED** [review]

File: website/docs/commands/basics.md [hunk 1]
```diff
@@ -174,12 +174,23 @@ scala-cli hello.zip
 
 ## Piping
 
-You can also pipe Scala code to `scala-cli` for execution:
-
-```bash
-echo 'println("Hello")' | scala-cli -
-# Hello
-```
+You can also pipe code to `scala-cli` for execution:
```
Comment (romanowski) [inline website/docs/commands/basics.md:177]: `echo 'println("Hello")' | scala-cli -` also works so we should include it in the scripts section

File: website/docs/commands/basics.md [hunk 2]
```diff
@@ -174,12 +174,23 @@ scala-cli hello.zip
...
+- scripts
+  ```bash
+  echo 'println("Hello")' | scala-cli _.sc
+  # Hello
+  ```
+- Scala code
+  ```bash
+  echo "@main def hello() = println(\"Hello\")" | scala-cli _.scala
```
Comment (romanowski) [inline website/docs/commands/basics.md]: We can mention that `echo "@main def hello() = println(\"Hello\")" | scala-cli _`

**Review by romanowski — COMMENTED** [review]
I've requested few clarifications, beside looks good

File: website/docs/commands/basics.md (hunk 1 above)
Comment (Gedochao) [reply website/docs/commands/basics.md:177]: there's actually 3 different aliases for the scripts wildcard (`-`, `_.sc` & `-.sc`), all of which are mentioned in the `Piping` guide. IMO it's enough to just give one example per section here.
I favored the underscore aliases for all 3 sections in `Basics` for consistency.

File: website/docs/commands/basics.md (hunk 2 above)
Comment (Gedochao) [reply website/docs/commands/basics.md]: same as with the scripts wildcard, there's also 3 aliases for the Scala code wildcard (`_`,`_.scala` and `-.scala`). I think such details should be left for the separate guide linked below.

Comment (alexarchambault) [inline website/docs/commands/basics.md]: Single quotes would make that simpler, wouldn't it?
```suggestion
  echo '@main def hello() = println("Hello")' | scala-cli _.scala
```

File: website/docs/commands/basics.md [hunk 3]
```diff
@@ -174,12 +174,23 @@ scala-cli hello.zip
...
+  # Hello
+  ```
+- Java code
+  ```bash
+  echo "class Hello { public static void main(String args[]) { System.out.println(\"Hello\"); } }" | scala-cli _.java
```
Comment (alexarchambault) [inline website/docs/commands/basics.md]: Same as above, single quotes makes that more readable IMO
```suggestion
  echo 'class Hello { public static void main(String args[]) { System.out.println("Hello"); } }' | scala-cli _.java
```

File: website/docs/guides/piping.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+---
+title: Piping
+sidebar_position: 23
+---
+
+# Piping
+
+Instead of passing paths to your sources, you can also pipe your code via standard input:
+```bash
+echo "@main def hello() = println(\"Hello\")" | scala-cli _
```
Comment (alexarchambault) [inline website/docs/guides/piping.md]: We could use single quotes here too.

File: website/docs/guides/piping.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+## Examples
+- scripts
+  ```bash
+  echo 'println("Hello")' | scala-cli _.sc
+  # Hello
+  ```
+- Scala code
+  ```bash
+  echo "@main def hello() = println(\"Hello\")" | scala-cli _.scala
```
Comment (alexarchambault) [inline website/docs/guides/piping.md]: Same as above.

File: website/docs/guides/piping.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  # Hello
+  ```
+- Java code
+  ```bash
+  echo "class Hello { public static void main(String args[]) { System.out.println(\"Hello\"); } }" | scala-cli _.java
```
Comment (alexarchambault) [inline website/docs/guides/piping.md]: Same as above.

File: website/docs/guides/piping.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  # Hello
+  ```
+
+## Mixing piped sources with on-disk ones
+It is also possible to pipe some code via standard input, while the rest of your code is on-disk.
+```bash
+echo "case class HelloMessage(msg: String)" > HelloMessage.scala
+echo "@main def hello() = println(HelloMessage(msg = \"Hello\").msg)" | scala-cli _ HelloMessage.scala
```
Comment (alexarchambault) [inline website/docs/guides/piping.md]: Same as above.

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**